### PR TITLE
Event index should be unique for tipsets

### PR DESCRIPTION
**Pull request by aarshkshah1992**
## Related Issues
Closes https://github.com/filecoin-project/lotus/issues/11630

## Proposed Changes
The event index for logs should be unique for all events in the same tipset to confirm with geth behaviour.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/events/filter/index.go [hunk 1]
```diff
@@ -356,6 +356,16 @@ func (ei *EventIndex) migrateToVersion4(ctx context.Context) error {
 		}
 	}
 
+	stmtEventIndexUpdate, err := tx.PrepareContext(ctx, "UPDATE event SET event_index = (SELECT COUNT(*) FROM event e2 WHERE e2.tipset_key_cid = event.tipset_key_cid AND e2.id <= event.id) - 1")
```
Comment (rvagg) [inline chain/events/filter/index.go:359]: I'm a bit worried about bundling this into the same migration. I'd be inclined to just introduce a v5 and stack them up just to be safe for anyone running on `master`. If we always move `ei.vacuumDBAndCheckpointWAL(ctx)` to the very last migration so we only do it once then the cost is pretty low.

Maybe this one isn't so critical but we've found from experience that people run on `master` and we don't have a good way of dealing with these migrations.

**Review by rvagg — COMMENTED** [review]
👌 yep, looks good, except for the migration concern, will take that discussion to Slack

File: chain/events/filter/index.go (hunk 1 above)
Comment (aarshkshah1992) [reply chain/events/filter/index.go:382]: @rvagg I agree that migrating to version 5 makes more sense here.

**Review by rvagg — APPROVED** [review]
Approving without running it. If this works on calibnet and you manually inspect the DB and see the migration worked then all good. The rest looks good.

**Comment by aarshkshah1992** [comment]
Tested this on calibnet

```
SELECT a.event_index, b.event_index FROM event a INNER JOIN event b ON a.tipset_key_cid = b.tipset_key_cid WHERE a.id != b.id LIMIT 120;
```

This query returns rows with duplicate event_indexes for master but does NOT do so for the current branch and returns sequential event_indices instead

```
4|0
4|1
4|2
4|3
4|5
4|6
4|7
4|8
4|9
4|10
4|11
4|12
4|13
4|14
4|15
5|0
5|1
5|2
5|3
5|4
5|6
5|7
5|8
5|9
5|10
5|11
5|12
5|13
5|14
5|15
6|0
6|1
6|2
6|3
6|4
6|5
6|7
6|8
```
......

**Pull request merged**